### ISSUE TITLE
Debug JavaScript with Safari stuck when press `Continue` or `Step over`

### ISSUE THREAD

**Issue by loongliu** [BabylonJS/BabylonNative#946]
**Describe the bug**
I tried to debug javascript with safari for iOS target and macOS target. 
The Safari connected to the JSC successfully.
I can see the console, view the source code, put a breakpoint, 
and the JSC stoped at the breakpoint.

But when I click `Continue` or `Step over`, the gui just stuck and nothing happened.
Tried debug for iOS target and macOS target, all the same situation.

**To Reproduce**

1. Build and run iOS target for default `experience.js`
2. Open safari and debug for the iOS target.
3. Put a breakpoint in `babylon.max.js` or  `experience.js`.
4. the breakpoint worked, and the JSC stoped.
5. But `Continue` or `Step over` does not has any effect.

**Expected behavior**
`Continue` or `Step over` should work.

**Screenshots**
![image](https://user-images.githubusercontent.com/15890184/140860755-b06b24c9-737d-4df4-966e-68572d327ad3.png)

**Other**
 - macOS 11.5.2
 - XCode 13.0 (13A233)
 - Safari 15.0 (16612.1.29.41.4, 16612)
 - iOS 13.8

**Comment by darraghjburke** [comment]
Hi, I was able to repro this. I'll do some investigation and see if I can find a solution. Hopefully it's not a Safari bug :)

**Comment by darraghjburke** [comment]
I tested on another version of Safari but it was still broken. Do you happen to know if there's a version of Safari/MacOS/BabylonNative where this did work?

We think it may have worked at one point, but we're not sure.

**Comment by bghgary** [comment]
I investigated this a tiny bit. It looks like calling `engine.runRenderLoop` somehow interferes with the JSC debugger. Just commenting out this section makes the JSC debugger work fine.

```javascript
    //engine.runRenderLoop(function () {
    //    scene.render();
    //});
```

I also tried just running the render loop with nothing in it, but this also breaks the JSC debugger.
```javascript
    engine.runRenderLoop(function () {
    //    scene.render();
    });
```

**Comment by bghgary** [comment]
The reason why this doesn't work is because the main thread is blocked and can't receive messages from the debugger. We need to rework the way rendering work to fix this. See #946. Closing as fixing the other issue should fix this.

**Comment by AJTech2002** [comment]
> The reason why this doesn't work is because the main thread is blocked and can't receive messages from the debugger. We need to rework the way rendering work to fix this. See [#946](https://github.com/BabylonJS/BabylonNative/issues/946). Closing as fixing the other issue should fix this.

Hey @bghgary this makes sense, this is still an issue and makes debugging very difficult - you've linked back to the same thread, is there another issue that's tracking this and are there any temporary work arounds in the meantime? 

Thanks :)

**Comment by bghgary** [comment]
Oops, I fixed the link. It should be #1416. Unfortunately, I don't know of a workaround. I know this is annoying. The fix is unfortunately a bigger change to how rendering works and will take some time before we can get to it.
